src/ansible_creator/config.py

```diff
@@ -23,14 +24,31 @@ class Config:
     collection: str = ""
     force: bool = False
     init_path: str = "./"
+    project: str = ""
+    scm_org: str = ""
+    scm_project: str = ""
 
     # TO-DO: Add instance variables for other 'create' and 'sample'
 
-    collection_name: str = field(init=False)
-    namespace: str = field(init=False)
+    collection_name: str = ""
+    namespace: str = ""
 
     def __post_init__(self: Config) -> None:
         """Post process config values."""
+        # Show CreatorError if the required collection name is not provided
+        if not self.collection and self.project == "collection":
+            msg = "The collection name is required when scaffolding a collection."
```

Comment:
Maybe use: "ansible-creator init: error: the argument 'collection' is required when scaffolding a collection"